include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,370 @@
+    //! Returns the number of elements in the container
+    EVE_FORCEINLINE std::size_t size()  const noexcept { return size_; }
+
+    //! Returns the number of elements that the container has currently allocated space for
+    EVE_FORCEINLINE std::size_t capacity()  const noexcept { return storage.capacity_; }
+
+    //! Requests the removal of unused capacity.
+    EVE_FORCEINLINE void shrink_to_fit() { grow(size(), size()); }
+
+    //! Checks if the container has no elements
+    EVE_FORCEINLINE bool empty() const noexcept { return !size(); }
+
+    //! @brief Increase the capacity of the vector to a value that's greater or equal to `n`.
+    //! If `n` is greater than the current capacity(), new storage is allocated and all iterators,
+    //! including the past-the-end iterator, and all references to the elements are invalidated.
+    //! Otherwise, the method does nothing.
+    //! @warning reserve() does not change the size of the vector.
+    //! @param n New capacity of the `eve::soa_vector`
+    void reserve(std::size_t n)
+    {
+      if(n > capacity())
+      {
+        // Grow capacity to n
+        auto const sz = size_;
+        grow(n, size());
+        size_ = sz;
+     }
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Modifiers
+    //! @{
+    //==============================================================================================
+    //! @brief Clear the contents of the container.
+    void clear() { size_ = 0; }
+
+    //! @brief Removes an element from the container.
+    //! Has the same invalidation semantics as std::vector.
+    //! end() iterator is not a valid pos.
+    iterator erase(const_iterator pos) { return erase(pos, pos+1); }
+
+    //! @brief Removes the elements in the range [first, last)
+    //! Empty range is OK, does nothing
+    iterator erase(const_iterator f, const_iterator l)
+    {
+      std::ptrdiff_t distance = f - cbegin();
+      std::ptrdiff_t sz       = l - f;
+
+      eve::algo::copy_backward( algo::as_range(l, cend())
+                              , algo::as_range(begin() + distance, end() - sz)
+                              );
+      size_ -= sz;
+      return begin() + distance;
+    }
+
+    //! @brief Appends the given element value to the end of the container.
+    //! If the new size() is greater than capacity() then all iterators and references (including
+    //! the past-the-end iterator) are invalidated. Otherwise only the past-the-end iterator is
+    //! invalidated.
+    //! @param value [Value](@eve::vectorizable) to append
+    EVE_FORCEINLINE void push_back(value_type const& value) noexcept
+    {
+      if(size_ == capacity())
```

Comment:
Do we want an unlikely attribute here?
We can also try to follow the llvm code structure to reduce the codegen on push back even more:
https://github.com/llvm/llvm-project/blob/bab2a4f2fbc183fb5899a915e37d0276a7662bd6/libcxx/include/vector#L1546

If we thing that's the right call.

Reply:
Reimplemented in a similar way